src/state/mod.rs

```diff
@@ -236,9 +237,117 @@ impl IndexerState {
         })
     }
 
-    /// Creates a new indexer state from a db instance
-    pub fn new_from_db(database_dir: &Path) -> anyhow::Result<Self> {
-        todo!("Restoring from db in {}", database_dir.display());
+    pub fn new_with_db(
+        root_block: Block,
+        transition_frontier_length: Option<u32>,
+        store: IndexerStore,
+    ) -> anyhow::Result<Self> {
+        let root_branch =
+            Branch::new_non_genesis(root_block.state_hash.clone(), root_block.blockchain_length);
+        let tip = Tip {
+            state_hash: root_block.state_hash.clone(),
+            node_id: root_branch.root.clone(),
+        };
+
+        Ok(Self {
+            mode: IndexerMode::Light,
+            phase: IndexerPhase::Watching,
+            canonical_tip: tip.clone(),
+            diffs_map: HashMap::new(),
+            best_tip: tip,
+            root_branch,
+            dangling_branches: Vec::new(),
+            indexer_store: Some(store),
+            transition_frontier_length: transition_frontier_length
+                .unwrap_or(MAINNET_TRANSITION_FRONTIER_K),
+            prune_interval: PRUNE_INTERVAL_DEFAULT,
+            canonical_update_threshold: CANONICAL_UPDATE_THRESHOLD,
+            blocks_processed: 0,
+            time: Instant::now(),
+            date_time: OffsetDateTime::now_utc(),
+        })
+    }
+
+    #[instrument]
+    pub fn restore_from_db(db: IndexerStore) -> anyhow::Result<Self> {
+        // TODO
+        // find best tip block in db (according to Block::cmp)
+        // go back at least 290 blocks (make this block the root of the root tree)
+        // Q: How to compute ledger? Should we require it to do quick sync?
+        // iterate over the db blocks, starting at the root, adding them to the state with add_block(..., false)
+
+        // find the best tip
+        debug!("finding best tip in the provided database");
```

Comment:
requiring this value will definitely reduce the compute time required to update the state, though it'll be a bit of effort to include it in the snapshot. I'd like to test the restoration without this value before making it required

Reply:
If you think that's the best way forward. I'd imagine it would be an overall easier process to include this ASAP, but you are the one writing the code. Do it to it!

Reply:
Wouldn't we only need to include the state hash of the `best_tip`? Would that really be complicated to add to our snapshot?